herbert_core/utilities/classes/@unique_references_container/unique_references_container.m

```diff
@@ -498,10 +498,10 @@
                 error('HERBERT:unique_references_container:incomplete_setup', ...
                     'global name unset');
             end
-            [glindex, ~] = self.global_container('value',self.global_name_).find_in_container(inobj);
+            [glindex, hash] = self.global_container('value',self.global_name_).find_in_container(inobj);
             if isempty(glindex)
                 glcont = self.global_container('value',self.global_name_);
-                [glcont,glindex] = glcont.add(inobj);
+                [glcont,glindex] = glcont.add_single_(inobj,[],hash);
                 if glindex == 0
                     % object was not added
```

Comment:
What is the "standard" signal that an object has not been added? Reading this is appears that empty `glindex` or `glindex == 0`, but that is two different signals. Is this correct?

Reply:
Very well spotted - it's an issue - but what is happening is that if isempty(glindex), the object has not been found and is added as a new object. If glindex==0 then nothing has been added to the container by add_single_, so the call has failed - not the same as not being found. However, a look at add_single_ shows that an object will always be added, so the if glindex==0 block is paranoia, and should be removed - I will do so.